File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int len = 0;
        ListNode *temp = head;
        while(temp){
            temp = temp->next;
            len++;
        }
        int pos = len - n;
        if(pos == 0){
            ListNode *newhead = head->next;
            delete head;
            return newhead;
        }
        temp = head;
        for(int i=0;i<pos-1;i++){
            temp = temp->next;
        }
        ListNode *del = temp->next;
        temp->next = del->next;
        delete del;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        count = 0
        temp = head
        while temp:
            count += 1
            temp = temp.next
        
        count -= n
        if count == 0:
            return head.next
        
        temp = head
        prev = head
        for j in range(count):
            prev = temp
            temp = temp.next
        
        prev.next = temp.next
        del(temp)
        return head